app/models/user.rb

```diff
@@ -27,5 +27,6 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   enum role: { student: 0, teacher: 1, admin: 2 }
+  validates :role, :name, presence: true
```

Comment:
can you test what happens if the role is an integer beyond the range? will it save?
if yes can you add a validation and a test?

Reply:
Sure, I'll work on that next

Reply:
@Naokimi Rails doesn't let you create new instances with enum values outside the defined ones, it throws an error
![image](https://user-images.githubusercontent.com/35246097/90200389-7cf7dc80-dd8c-11ea-877e-553551fdd47f.png)